packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts

```diff
@@ -376,41 +604,120 @@ export class AnchoredRegion extends FASTElement {
-            if (entry.target === this.region) {
-                this.handlePositionerCollision(entry, entries.length === 1);
-                positionerRect = entry.boundingClientRect;
+    private isValidIntersection = (entry: IntersectionObserverEntry): boolean => {
+        return (
+            entry.rootBounds !== null &&
+            this.viewportElement !== null &&
+            Math.round(entry.rootBounds.height) ===
+                Math.round(this.viewportElement.clientHeight)
+        );
+    };
+
+    /**
+     *  Handle intersections
+     */
+    private handleIntersection = (entries: IntersectionObserverEntry[]): void => {
+        if (this.viewportElement === null || this.anchorElement === null) {
+            return;
+        }
+
+        let regionRect: DOMRect | ClientRect | null = null;
+
+        if (!this.initialLayoutComplete) {
+            if (entries.length !== 2) {
+                return;
+            }
+
+            if (
+                (!this.isValidIntersection(entries[0]) ||
+                    !this.isValidIntersection(entries[1])) &&
+                this.useGbcr !== "never"
+            ) {
+                regionRect = this.applyNoIntersectionMode();
             } else {
-                this.handleAnchorCollision(entry);
+                regionRect = this.applyIntersectionEntries(entries);
             }
-        });
 
-        if (this.viewportElement !== null) {
-            this.viewportScrollTop = this.viewportElement.scrollTop;
-            this.viewportScrollLeft = RtlScrollConverter.getScrollLeft(
-                this.viewportElement,
-                this.currentDirection
-            );
+            if (regionRect !== null) {
+                this.updateRegionOffset(regionRect);
+            }
+
+            this.requestLayoutUpdate();
+        } else {
+            if (!this.noIntersectionMode) {
+                if (entries.length === 2) {
+                    this.applyIntersectionEntries(entries);
+                    this.requestLayoutUpdate();
+                } else {
+                    this.update();
+                }
+            }
         }
-        if (entries.length === 2 && positionerRect !== null) {
-            this.updatePositionerOffset(positionerRect);
+    };
+
+    /**
+     *  use getBoundingClientRect when intersection observer fails
+     */
+    private applyNoIntersectionMode = (): ClientRect | null => {
+        this.noIntersectionMode = true;
+        let regionRect: DOMRect | ClientRect | null = this.getBoundingClientRect();
+
+        if (
+            this.viewportElement === null ||
+            this.anchorElement === null ||
+            regionRect === null
+        ) {
+            return null;
         }
-        this.requestLayoutUpdate();
+
+        this.regionDimension = {
+            height: regionRect.height,
+            width: regionRect.width,
+        };
+
+        // todo: scomea, try to avoid getting viewportRect when it is document root
```

Comment:
Is this still actionable?

Reply:
Yes, I think there is potential perf optimization there, but deferring on it until later.